src/masternodes/mn_checks.cpp

```diff
@@ -91,8 +91,7 @@ bool HasAuth(CTransaction const & tx, CCoinsViewCache const & coins, CScript con
 {
     for (auto input : tx.vin) {
         const Coin& coin = coins.AccessCoin(input.prevout);
-        assert(!coin.IsSpent());
-        if (coin.out.scriptPubKey == auth)
+        if (!coin.IsSpent() && coin.out.scriptPubKey == auth)
```

Comment:
Could this affect consensus

Reply:
No. Either way, if `!coin.IsSpent()` is false then the function is not going to return true.